freud/_locality.pxd

```diff
@@ -101,31 +101,30 @@ cdef extern from "LinkCell.h" namespace "freud::locality":
         unsigned int begin()
 
     cdef cppclass LinkCell(NeighborQuery):
-        LinkCell()
-        LinkCell(const freud._box.Box &, float) except +
-        LinkCell(const freud._box.Box &, float,
-                 const vec3[float]*, unsigned int) except +
-
-        setCellWidth(float) except +
-        updateBox(const freud._box.Box &) except +
+        LinkCell() except +
+        LinkCell(const freud._box.Box &,
+                 float,
+                 const vec3[float]*,
+                 unsigned int) except +
         const vec3[unsigned int] computeDimensions(
             const freud._box.Box &,
             float) const
         unsigned int getNumCells() const
         float getCellWidth() const
         unsigned int getCell(const vec3[float] &) const
         IteratorLinkCell itercell(unsigned int) const
-        vector[unsigned int] getCellNeighbors(unsigned int) const
```

Comment:
Since `m_cell_neighbors` is basically just a convenient cache, I think it would make sense to leave this function const and just make that object `mutable`.